test/e2e/kcp_upgrade.go

```diff
@@ -92,20 +92,34 @@ func KCPUpgradeSpec(ctx context.Context, inputGetter func() KCPUpgradeSpecInput)
 			WaitForMachineDeployments:    input.E2EConfig.GetIntervals(specName, "wait-worker-nodes"),
 		})
 
-		By("Upgrading Kubernetes, DNS, kube-proxy, and etcd versions")
+		By("Upgrading Kubernetes & kube-proxy to 1.17, CoreDNS to 1.6.5, and etcd to 3.4.3")
 		framework.UpgradeControlPlaneAndWaitForUpgrade(ctx, framework.UpgradeControlPlaneAndWaitForUpgradeInput{
 			ClusterProxy: input.BootstrapClusterProxy,
 			Cluster:      cluster,
 			ControlPlane: controlPlane,
-			//Valid image tags for v1.17.2
+			// Valid image tags for v1.17.2
 			EtcdImageTag:                "3.4.3-0",
-			DNSImageTag:                 "1.6.6",
+			DNSImageTag:                 "1.6.5",
 			KubernetesUpgradeVersion:    "v1.17.2",
 			WaitForMachinesToBeUpgraded: input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 			WaitForDNSUpgrade:           input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 			WaitForEtcdUpgrade:          input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 		})
 
+		By("Upgrading Kubernetes & kube-proxy to 1.18, CoreDNS to 1.6.7, and etcd to 3.4.3")
+		framework.UpgradeControlPlaneAndWaitForUpgrade(ctx, framework.UpgradeControlPlaneAndWaitForUpgradeInput{
+			ClusterProxy: input.BootstrapClusterProxy,
+			Cluster:      cluster,
+			ControlPlane: controlPlane,
+			// Valid image tags for v1.18.2
+			EtcdImageTag:                "3.4.3-0",
```

Comment:
If I'm not wrong, having hard coded EtcdImageTag and DNSImageTag can lead to problems in future when KubernetesVersion changes
@sedefsavas ^^?